src/valkeymodule.h

```diff
@@ -1649,6 +1691,19 @@ VALKEYMODULE_API int (*ValkeyModule_RdbSave)(ValkeyModuleCtx *ctx,
                                              ValkeyModuleRdbStream *stream,
                                              int flags) VALKEYMODULE_ATTR;
 
+VALKEYMODULE_API int (*ValkeyModule_RegisterScriptingEngine)(ValkeyModuleCtx *ctx,
+                                                             const char *engine_name,
+                                                             void *engine_ctx,
+                                                             ValkeyModuleScriptingEngineCreateFunctionsLibraryFunc create_functions_library_func,
+                                                             ValkeyModuleScriptingEngineCallFunctionFunc call_function_func,
+                                                             ValkeyModuleScriptingEngineGetUsedMemoryFunc get_used_memory_func,
+                                                             ValkeyModuleScriptingEngineGetFunctionMemoryOverheadFunc get_function_memory_overhead_func,
+                                                             ValkeyModuleScriptingEngineGetEngineMemoryOverheadFunc get_engine_memory_overhead_func,
+                                                             ValkeyModuleScriptingEngineFreeFunctionFunc free_function_func) VALKEYMODULE_ATTR;
```

Comment:
I think it makes sense to use a struct to group the callbacks because of the extensibility argument. I will change it.

Reply:
> Do we want to add an optional callback to kill a script or function? It can be invoked by SCRIPT KILL or FUNCTION KILL.

I'll address this in the followup PR that implements EVAL.